fpga_interchange/site_router.cc

```diff
@@ -1133,6 +1140,57 @@ static void block_cluster_wires(SiteArch *site_arch)
     }
 }
 
+// Reserve site ports that are on dedicated rather than general interconnect
+static void reserve_site_ports(SiteArch *site_arch)
+{
+    const Context *ctx = site_arch->site_info->ctx;
+    site_arch->blocked_site_ports.clear();
+    for (PipId in_pip : site_arch->input_site_ports) {
+        pool<NetInfo*, hash_ptr_ops> dedicated_nets;
+        const int max_iters = 1000;
```

Comment:
Might this be too high? To my understanding, the majority of the cases end up in reaching `max_iters`, unless there is a dedicated net and `max_iters` won't be reached.

Given that dedicated nets are quite short and do not require too many hops, would it make sense to reduce `max_iters`?

Reply:
sure, I'll reduce it it to 100, that'll be fine for now. if we hit problems with that we can look at better heuristics